python/seldon_core/metadata.py

```diff
@@ -146,38 +128,37 @@ def validate_model_metadata(data: Dict) -> Dict:
     Notes
     -----
 
-    Read data from json and create ModelMetadataValidator and MetadataTensorValidator objects.
-    This function reads data in safe manner from json: validation and exceptions
-    will happen in ModelMetadataValidator and MetadataTensorValidator classes.
-
-    SeldonInvalidMetadataError has been chosen for exception as validation mostly depend on having
-    a correct type for specific components.
+    Read data from json and validate against v1 or v2 metadata schema.
+    SeldonInvalidMetadataError exception will be raised if validation fails.
     """
     if MODEL_IMAGE is not None:
         image_name, image_version = split_image_tag(MODEL_IMAGE)
     else:
         image_name, image_version = "", ""
-    name = data.get("name", image_name)
-    versions = data.get("versions", [image_version])
-    platform = data.get("platform", "")
+
+    default_meta = {
+        "apiVersion": "v2",
+        "name": image_name,
+        "versions": [image_version],
+        "platform": "",
+        "inputs": [],
+        "outputs": [],
+    }
+
+    data = {**default_meta, **data}
+    v = data.get("apiVersion", "v2")
+
+    if v == "v1":
+        schema = V1_SCHEMA
+    elif v == "v2":
+        schema = V2_SCHEMA
+    else:
+        raise SeldonInvalidMetadataError(f"Unknown metadata schema: {v}")
 
     try:
-        inputs = [
-            MetadataTensorValidator(x.get("name"), x.get("datatype"), x.get("shape"))
-            for x in data.get("inputs", [])
-        ]
-
-        outputs = [
-            MetadataTensorValidator(x.get("name"), x.get("datatype"), x.get("shape"))
-            for x in data.get("outputs", [])
-        ]
-    except (AttributeError, TypeError):
-        raise SeldonInvalidMetadataError(
-            "Model metadata inputs and outputs must be sequence of dictionaries."
-        )
-
-    meta = ModelMetadataValidator(
-        name=name, versions=versions, platform=platform, inputs=inputs, outputs=outputs
-    )
-
-    return meta.to_dict()
+        validate(data, schema)
```

Comment:
Nice! That looks much cleaner now!

Reply:
That's the point, leave the code easier to read and follow on every improvement ; - )